hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/FileIndex.java

```diff
@@ -80,6 +80,12 @@ public static FileIndex instance(Path path, Configuration conf, RowType rowType)
     return new FileIndex(path, conf, rowType);
   }
 
+  public static FileIndex instance(Path path, Configuration conf, RowType rowType, List<ResolvedExpression> filters) {
+    FileIndex fileIndex = instance(path, conf, rowType);
```

Comment:
Nice catch ~ Should we also keep this method: `instance(Path path, Configuration conf, RowType rowType)` ?

Reply:
Is is a really good question
Quick answer -- yes we should keep because the method is used in `IncrementalInputSplits` without filter at all.
But why there is no a filter in `IncrementalInputSplits`. Looks like `FileIndex` useless in this way. I mean data skipping feature skips nothing
Mb we should pass filters from `HoodieTableSource` here:
```java
case FlinkOptions.QUERY_TYPE_INCREMENTAL:
  IncrementalInputSplits incrementalInputSplits = IncrementalInputSplits.builder()
      .conf(conf)
      .path(FilePathUtils.toFlinkPath(path))
      .rowType(this.tableRowType)
      .maxCompactionMemoryInBytes(maxCompactionMemoryInBytes)
      .requiredPartitions(getRequiredPartitionPaths()).build();
```
I think it is not part of this fix related to `copy()` by `HepPlanner`
WDYT about the separate ticket/PR for this problem if the one exists?:)

Reply:
Agree, we can fix it in another PR.

Reply:
And another solution is we add a `FileIndex` param in the constructor so that there is no need to re-construct it again and again.

Reply:
Thanks, you are right
- Added `FileIndex` param in the constructor
- Replaced `this.filters` with local variable in `applyFilters(filters)`
- Removed field `List<ResolvedExpression> filters` as no need anymore
- Reverted `instance(path, conf, rowType, filters)` as no need anymore